matplotlibrc.template

```diff
@@ -414,6 +414,7 @@ backend      : $TEMPLATE_BACKEND
 #legend.scatterpoints : 1        # number of scatter points
 #legend.markerscale   : 1.0      # the relative size of legend markers vs. original
 #legend.fontsize      : medium
+#legend.titlesize     : medium    # fontsize of the legend title
```

Comment:
This is a breaking change? Because the legend fontsize *used* to be the axes fontsize, so if someone changed the axes fontsize they may expect the legend title to also change size.  So... I think it needs an API entry as well.

Reply:
I see what you're saying. On unpatched matplotlib:
```
In [28]: mpl.rcParams.update({'font.size': 22})
In [29]: ax = plt.subplot(111);ax.scatter(np.arange(0,5), np.arange(0,5), label='One')
In [30]: ax.legend(fontsize=18, title='My Great Legend')
In [31]: ax.get_legend().get_title().get_fontsize()
Out[31]: 22.0
```

With this patch that new font size would be this default value. Suggestions for how to proceed? Just an entry in the API doc?

Reply:
Two choices, I guess.  Either just add an API change.  I'd *personally* be fine w/ that.  

Or, if you want to be fancy, the default could be `"None"`, in which case it could be made to just use the axes fontsize. If the user specifies a fontsize, then that gets used instead.  

Given that there was no test that broke, I assume the default is 'medium'.  OTOH, its a bit of a pain to set all the different fontsizes if you want them to be bigger, and this one is particularly obscure.  So maybe "None" is the way to go?  I don't feel strongly either way.

I'm not sure off the top of my head how "None" gets parsed in the rcParams, but there are a few entries with a None entry, so I imagine you could figure it out.

Reply:
I looked again and there is no validate_fontsize_or_none validation and I do t see why you should do it here.  I’d say just add an API doc.

Reply:
I have added a first pass at an API doc.

Reply:
Another option here is to default to 'inherit' which falls back to not passing anything into `set_title` (we do this a few other places).

Reply:
@tacaswell I could do that. Do we also want this to be a kwarg?

Reply:
@Raab70  if you take a look at `rcsetup.py` you'll see how `validate_color_or_inherit` works.

Reply:
I am fine either way about adding a kwarg.  On one hand, it would be reasonable to expect an kwarg on the other hand `legend` already has too many kwrags.

Reply:
In that case I would go with your comment above to keep the clutter down for something that is not likely to be used very often. 
>If and when someone asks for per-call-to-legend control of this we will sort out how to do kwargs with this.